### DISCUSSION TITLE
Add boost-like Integer Type Selection

### DISCUSSION THREAD

**Discussion by TomSaw** [modm-io/modm#718] [Ideas]
I quite often need a boost like [Integer Type Selection](https://www.boost.org/doc/libs/1_47_0/libs/integer/doc/html/boost_integer/integer.html). Currently for making `modm::color` pure generic.

Shall i port this or is there a another plan in the shelf?

**Comment by TomSaw** [comment]
Something like this:

template <unsigned int Bits>
struct smallest_uint
{
	using type = std::conditional_t<
		(Bits <= 8), uint8_t, std::conditional_t<
		(Bits <= 16), uint16_t, std::conditional_t<
		(Bits <= 32), uint32_t, std::enable_if_t<
		(Bits <= 64), uint64_t>
	>>>;
};

template <unsigned int Bits>
using smallest_uint_t = typename small_uint<Bits>::type;

**Comment by salkinium** [comment]
Not sure if I'm ready to integrate boost, so I'd port it.

**Comment by TomSaw** [comment]
0kay i've kept it simple and added `/modm/math/utils/integer_traits.hpp` to #665 . It includes additional magic i needed for fully generic `modm:ui:color`.
